### PR TITLE
Fix MinGW compilation on drives besides C:

### PR DESCRIPTION
**Pull request by Bobo1239**
Also removed some unnecessary (as far as I can see) indirection.

Fixes https://github.com/alexcrichton/curl-rust/issues/189.

### CODE REVIEW

File: src/multi.rs [hunk 1]
```diff
@@ -341,10 +341,6 @@ impl Multi {
     /// error code from the transfer or just the fact that a transfer is
     /// completed. More details on these should be written down as well.
     pub fn messages<F>(&self, mut f: F) where F: FnMut(Message) {
-        self._messages(&mut f)
-    }
-
-    fn _messages(&self, mut f: &mut FnMut(Message)) {
```
Comment (alexcrichton) [inline src/multi.rs]: Oh could this be left as-is? The intention here is to reduce monomorphization burden on callers

**Comment by alexcrichton** [comment]
Looks great, thanks!

File: src/multi.rs (hunk 1 above)
Comment (Bobo1239) [reply src/multi.rs]: Ah I see, TIL. Removed the second commit.

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**